rp_asos/tasks.py

```diff
@@ -212,108 +176,81 @@ def get_reminders_for_date(
+                project, hospital, hospital_patient_records
             )
 
-            if (
-                hospital_screening_records
-                and hospital_screening_records[0][screening_field]
-            ):
-                patient_count = int(
-                    hospital_screening_records[0][screening_field]
-                )
-
-                # Check count
-                if patient_count != len(hospital_patient_records):
-                    messages[hospital][date].append(
-                        "Not all patients captured.({}/{})".format(
-                            len(hospital_patient_records), patient_count
-                        )
-                    )
-
-                # check status
-                for patient in hospital_patient_records:
-                    if (
-                        patient.get("pre_operation_status")
-                        != PatientRecord.COMPLETE_STATUS
-                        or patient.get("post_operation_status")
-                        != PatientRecord.COMPLETE_STATUS
-                    ):
-                        messages[hospital][date].append(
-                            "{}: {} preoperative, {} postoperative fields missing".format(
-                                patient["record_id"],
-                                len(patient["missing_pre_op_fields"]),
-                                len(patient["missing_post_op_fields"]),
-                            )
-                        )
-            else:
-                messages[hospital][date].append(
-                    "No screening records found.({})".format(date)
-                )
-
-        self.save_patient_records(project, patient_records, date)
-
-        return messages
-
-    def send_reminders(self, messages, rapidpro_client, org):
-        for hospital, msgs in messages.items():
-            reminders = []
-            for date, hosp_msgs in msgs.items():
-                if hosp_msgs:
-                    reminders.append(
-                        "\nFor surgeries registered on {}:".format(
-                            date.strftime("%d %B %Y")
-                        )
-                    )
-                for msg in hosp_msgs:
-                    reminders.append(msg)
-
-            if reminders:
-                hospital.send_message(reminders)
-
-    def run(self, args, **kwargs):
-        project_id = args[0]
-        tz_code = args[1]
-
+    def run(self, project_id, tz_code, **kwargs):
         project = Project.objects.prefetch_related("hospitals").get(
             id=project_id
         )
 
-        screening_client = project.get_redcap_client()
-        patient_client = project.get_redcap_crf_client()
-        rapidpro_client = project.org.get_rapidpro_client()
-
-        metadata = self.get_metadata(patient_client)
-        required_fields = self.get_required_fields(metadata)
+        message_template = (
+            "Daily data update for {}:\n"
```

Comment:
I think it might make help to understand the template better by using keywords, i.e.
```
"Daily data update for {name}:\n"
```
and then when using the template, using keyword args:
```
message_template.format(
    name=hospital.name,
    . . . 
)
```

I'm just a bit concerned that using order for this many args leaves a fair bit of room for error

Reply:
Good suggestion!